### PR TITLE
Rustler 0.30 & latest Elixir/Erlang

### PR DESCRIPTION
**Pull request by RoyalIcing**
Hey! I saw there’s a new version of Rustler, so here’s an update to it. [I believe it’s just a bump with minor changes.](https://github.com/rusterlium/rustler/releases/tag/rustler-0.30.0)

I can make it in a separate PR if you prefer: there’s also updates to Elixir/Erlang so I’ve updated those too.

### CODE REVIEW

File: native/wasmex/src/environment.rs [hunk 1]
```diff
@@ -47,7 +47,7 @@ fn link_import(
     let import_tuple = tuple::get_tuple(definition)?;
 
     let import_type = import_tuple
-        .get(0)
+        .first()
```
Comment (tessi) [inline native/wasmex/src/environment.rs:50]: ❤️

**Comment by tessi** [comment]
@RoyalIcing this looks great! thanks for taking care! ❤️ 

there is just that weird dialyzer error. and just staring at the error output I sometimes wonder if it's worth having it. I cheer for elixirs type system to advance fast :)

**Comment by RoyalIcing** [comment]
@tessi I'll have a look at the error, I missed that. Yeah I much prefer first-party type systems!

File: native/wasmex/src/environment.rs (hunk 1 above)
Comment (RoyalIcing) [reply native/wasmex/src/environment.rs:50]: It was the Rust linter’s idea :)

**Comment by tessi** [comment]
whoar, I guess I'll throw out dialyzer. the benefit it brings doesn't feel big enough compared to the burden of maintenance it introduces.

**Comment by RoyalIcing** [comment]
> whoar, I guess I'll throw out dialyzer. the benefit it brings doesn't feel big enough compared to the burden of maintenance it introduces.

Yeah it only seems to be happening with the latest Elixir? It's not saying which line the problem is coming from so it's difficult to debug. I couldn't see a function matching the erring pattern at first glance.

**Comment by tessi** [comment]
alright, dialyzer is gone for good. this PR is green. again. Thanks a ton @RoyalIcing 💜

**Pull request merged**